### ISSUE TITLE
Math mutator: Don't mutate string concatenation '+'

### ISSUE THREAD

**Issue by DiegoRBaquero** [TheSoftwareDesignLab/mutode#10]

**Comment by simondel** [comment]
Why not? It's valid JavaScript.

**Comment by DiegoRBaquero** [comment]
Valid JavaScript but results in NaN. Easily caught while coding, when failing to see the expected text or by a linter. Saves many mutators when there are many concatenations -> saves time.

**Comment by simondel** [comment]
:thumbsup: your call. You could also keep it in to encourage devs to switch to template strings.

In the case of Stryker, the only complaints we got from mutating string concatenation was from TypeScript users and we fixed that by using the TypeScript compiler together with the projects tsconfig.json to determine which code would cause a transpiler error.

**Comment by DiegoRBaquero** [comment]
This I like
> You could also keep it in to encourage devs to switch to template strings.

How did you mutate string concat? + to - too?

**Comment by simondel** [comment]
Yes. We don't look at the types of the left or right operand in binary expressions. We simply change the plus token into a minus token.

With JavaScript code, the mutant will actually get tested. With TypeScript it will first be ran though the TypeScript compiler which will (in most cases) cause an error. Mutants that cause a transpilation error are not tested and not counted in the score.
